plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RendererCreatorImpl.java

```diff
@@ -499,12 +499,14 @@ public SelectionLayer findSelectionLayer( ILayer targetLayer ) {
         } catch (IOException e) {
             return null;
         }
-        for( Layer layer : getLayers() )
-            if (layer instanceof SelectionLayer)
-                if (((SelectionLayer) layer).getWrappedLayer() == targetLayer)
-                    return (SelectionLayer) layer;
+        synchronized (getLayers()) {
```

Comment:
We added interface ISynchronizedEListIteration with method to synchronize iteration through a list of Layers. See #283 as an example..
I guess it would help here either..